google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployAction.java

```diff
@@ -167,7 +155,20 @@ public void onTextAvailable(ProcessEvent event, Key outputType) {
     public void processTerminated(final ProcessEvent event) {
       try {
         if (event.getExitCode() == 0) {
-          callback.succeeded(new DeploymentRuntimeImpl(deploymentOutput.toString(), version));
+
+          // Parse JSON output to retrieve service/version info.
+          DeployOutput deployOutput = null;
+          try {
+            deployOutput = parseDeployOutput(deploymentOutput.toString());
+          } catch (JsonParseException e) {
+            getLoggingHandler().print(
+                GctBundle.message("appengine.deployment.version.extraction.failure") + "\n");
+            logger.warn("Could not retrieve service/version info of deployed application", e);
+          }
+
+          callback.succeeded(
+              new AppEngineDeploymentRuntime(project, appEngineHelper, getLoggingHandler(),
+                                             deployOutput.getService(), deployOutput.getVersion()));
```

Comment:
wouldn't this throw an NPE if the parsing fails since deployOutput will be null?